core-bundle/src/Resources/contao/library/Contao/Dbafs.php

```diff
@@ -361,16 +361,20 @@ public static function copyResource($strSource, $strDestination)
 
 			if ($objFiles !== null)
 			{
+				$arrMapper = array();
+
 				while ($objFiles->next())
 				{
 					/** @var FilesModel $objNew */
 					$objNew = clone $objFiles->current();
 
-					$objNew->pid    = $objNewFile->uuid;
+					$objNew->pid    = $arrMapper[$objFiles->uuid] ?? $objNewFile->uuid;
```

Comment:
```suggestion
					$objNew->pid    = $arrMapper[$objFiles->pid] ?? $objNewFile->uuid;
```
we have to check for the old `pid` I guess?